src/Jss.js

```diff
@@ -74,22 +75,33 @@ export default class Jss {
    * @see createRule
    * @api public
    */
-  createRule(selector, style, options) {
-    // Enable rule without selector.
-    if (typeof selector == 'object') {
+  createRule(name, style, options) {
+    // Enable rule without name for inline styles.
+    if (typeof name === 'object') {
       options = style
-      style = selector
-      selector = null
+      style = name
+      name = null
     }
-    const rule = createRule(selector, style, {
+
+    options = {
       jss: this,
       Renderer: findRenderer(options),
       ...options
-    })
-    this.plugins.run(rule)
+    }
+
+    const rule = this.factory.get(name, style, options)
+
+    if (options.applyPlugins !== false) {
+      this.plugins.handleRule(rule)
+    }
+
     return rule
   }
 
+  registerRuleClass(name, cls) {
```

Comment:
Like, Ideally the plugin API should support the following functionality:

- Plugin can parse the rule name before anything is created
- If it's a global scope selector, the plugin can set the selector as is necessary
- If it is a global scope nested object of rules, the plugin can set the rule type, so `jss` knows it is not a regular CSS rule
- When the nested rules are parsed, the plugin can check if the parent is a globally scoped rule and manipulate the selectors as is necessary